provider/resource_keycloak_attribute_importer_identity_provider_mapper.go

```diff
@@ -71,12 +71,30 @@ func getAttributeImporterIdentityProviderMapperFromData(data *schema.ResourceDat
 			return nil, fmt.Errorf(`provider.keycloak: keycloak_attribute_importer_identity_provider_mapper: %s: "claim_name": should be set for %s identity provider`, data.Get("name").(string), identityProvider.ProviderId)
 		}
 		rec.Config.Claim = data.Get("claim_name").(string)
+
+	} else if isAttributeImportEnabledSocialProvider(identityProvider.ProviderId) {
```

Comment:
I think if there are a bunch of these social providers where these are required fields, it would make sense to add them as top level attributes.  if there's a conflict with an existing attribute name, we can try and figure that out.

Reply:
Hey @mrparkers, sorry about the delay in responding, got caught up on another project. So the reason why I used the extra config block for the `userAttribute` field is because as you hinted, there are conflicts with existing fields.

Both SAML and OIDC clients refer to the user attribute in the payload as `config."user.attribute"` whereas social identity providers refer to it as `config.userAttribute`. I tried using the top level `user_attribute` at first, but it wasn't working because it's marshalled as `user.attribute` in `IdentityProviderMapperConfig`. So then I tried adding another field in that struct that would be marshalled as `userAttribute`, but that was generating inconsistencies between creates, updates and reads (I would have to dig a little more since it's been more than a month). Any suggestion you have is more than welcome, I'm not that used to developing terraform providers.

To feed the discussion, below are the payloads I'm seeing when creating attribute importer mappers.

## OIDC Client

```json
{
  "identityProviderAlias": "oidc",
  "config": {
    "syncMode": "INHERIT",
    "claim": "remoteClaim",
    "user.attribute": "userProperty"
  },
  "name": "attribute-mapper-name",
  "identityProviderMapper": "oidc-user-attribute-idp-mapper"
}
```

## SAML Client

```json
{
  "identityProviderAlias": "saml",
  "config": {
    "syncMode": "INHERIT",
    "attributes": "[]",
    "attribute.name": "remoteAttributeName",
    "attribute.friendly.name": "someFriendlyName",
    "user.attribute": "userProperty"
  },
  "name": "attribute-importer-name",
  "identityProviderMapper": "saml-user-attribute-idp-mapper"
}
```

## Social Provider

```json
{
  "identityProviderAlias": "github",
  "config": {
    "syncMode": "INHERIT",
    "jsonField": "remoteClaim",
    "userAttribute": "userProperty"
  },
  "name": "attribute-importer-name",
  "identityProviderMapper": "github-user-attribute-mapper"
}
```